storageservice/modules/files/file_client.bal

```diff
@@ -376,71 +376,82 @@ public isolated client class FileClient {
-                                     returns @display {label: "Response"} byte[]|error {
-        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH 
-            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);    
+    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName,
+                                    @display {label: "Azure File Name"} string fileName,
+                                    @display {label: "Azure Directory Path"} string? azureDirectoryPath = (),
+                                    @display {label: "Specified Byte Range"} ContentRange? range = ())
+                                    returns @display {label: "Response"} byte[]|error {
+        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH
+            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);
         http:Request request = new;
+        if (range != ()) {
+            string rangeHeader = "bytes=" + range.startByte.toString() + "-" + range.endByte.toString();
+            request.setHeader(X_MS_RANGE, rangeHeader);
+        }
         if (self.azureConfig.authorizationMethod == ACCESS_KEY) {
-            map<string> requiredURIParameters = {}; 
-            string resourcePathForSharedkeyAuth = azureDirectoryPath is () ? (fileShareName + SLASH + fileName) 
+            map<string> requiredURIParameters = {};
+            string resourcePathForSharedkeyAuth = azureDirectoryPath is () ? (fileShareName + SLASH + fileName)
                 : (fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);
-            AuthorizationDetail  authorizationDetail = {
+            AuthorizationDetail authorizationDetail = {
                 azureRequest: request,
                 azureConfig: self.azureConfig,
                 httpVerb: http:HTTP_GET,
                 resourcePath: resourcePathForSharedkeyAuth,
                 requiredURIParameters: requiredURIParameters
             };
-            check prepareAuthorizationHeaders(authorizationDetail);      
+            check prepareAuthorizationHeaders(authorizationDetail);
         } else {
-            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1)); 
+            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1));
         }
         map<string> headerMap = populateHeaderMapFromRequest(request);
         http:Response response = check self.httpClient->get(requestPath, headerMap);
-        byte[] payload = check response.getBinaryPayload();
-        return payload;
+        boolean isFileTypeAvailable = response.hasHeader(X_MS_TYPE);
+        if (isFileTypeAvailable == true) {
+            byte[] payload = check response.getBinaryPayload();
```

Comment:
We can directly return here?

Reply:
This was the fix we added, Earlier we returned directly but for an incorrect file path, the error message also comes in the response body as XML file type with 200 status code. therefore, by using this workaround we can get rid of that issue